### PR TITLE
Added a delete action to S3 buckets

### PR DESCRIPTION
**Pull request by bigkraig**

### CODE REVIEW

File: c7n/resources/s3.py [hunk 1]
```diff
@@ -514,9 +514,16 @@ def add(self, keys):
         self.count += len(keys)
         if self.fh is None:
             return
-        self.fh.write(json.dumps(keys))
+        self.fh.write(json.dumps(keys, cls=DateEncoder))
         self.fh.write(",\n")
 
+class DateEncoder(json.JSONEncoder):
```
Comment (kapilt) [inline c7n/resources/s3.py]: we have one of these in utils.py

File: c7n/resources/s3.py [hunk 2]
```diff
@@ -1041,3 +1048,68 @@ def process_resource_set(self, resource_set, tags):
...
+        return results
+
+    def process_chunk(self, batch, bucket):
+        s3 = bucket_client(
+            local_session(self.manager.session_factory), bucket)
+        b = bucket['Name']
+
+        keys = [{'Key': key['Key']} for key in batch]
+        objects = {'Objects': keys}
+        response = s3.delete_objects(Bucket=b, Delete=objects)
```
Comment (kapilt) [inline c7n/resources/s3.py]: there's some minor variation here depending on whether the bucket is versioned or not in which case the key should also include the VersionId, the scan base class auto switches to iterating over object versions on versioned buckets, so its just a minor of including the VersionId from the key dict if the bucket is versioned.

**Comment by kapilt** [comment]
Thanks!, looking good, some minors comments, One other item is signing our CLA as described at the bottom of the readme.
https://github.com/capitalone/cloud-custodian

**Comment by bigkraig** [comment]
@kapilt Got those changes made and i've signed the CLA 👍

**Comment by kapilt** [comment]
looks good, i merged your other pull request though, and this needs an update before i can merge.

**Comment by kapilt** [comment]
merging as is, we'll need to backfill unit tests. Thanks!

**Comment by kapilt** [comment]
fwiw, i'm backfilling on unit tests, going to rename the 'empty' parameter to 'remove-contents' as the former is easily mistaken as a condition for the action (ie. empty?) rather than the imperative that it conrols, thinking about 'remove-contents'

**Pull request merged**